Merge new build_multinetwork parser and tests

skylerreid opened the pull request:
closed old pr and consolidated changes in this branch since the tests are already started here. changed internal function name from `_create_time_series_block_no_interp` to `_create_tsb` as per Russell's previous comment.

rb004f commented:
@skylerreid there are 9 tests that are failing with this PR.  These will need be resolved before we can merge.  Thanks!

skylerreid commented:
summary of updates: added new functions for parsing m+static and m+transient files. fixed matgas writer. added function for solution hints in both transient and static files. updates to tests everywhere. numerous bug fixes throughout the code.
